v2/spiffebundle/set.go

```diff
@@ -0,0 +1,45 @@
+package spiffebundle
+
+import (
+	"crypto"
+	"crypto/x509"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+// Set is a set of bundles, keyed by trust domain.
+type Set struct{}
+
+// NewSet creates a new, empty set.
+func NewSet() *Set {
+	panic("not implemented")
+}
+
+// Insert inserts a new bundle into the set. If a bundle already exists for the
+// trust domain, the existing bundle is replaced.
+func (set *Set) Insert(trustDomain spiffeid.TrustDomain, bundle *Bundle) {
```

Comment:
When I originally read the examples that consume this, I mistook "set" to mean a bundle (because a bundle has many keys in it), and mistook Insert for "insert new keys into this bundle".

I'm not sure I understand what Set buys us here? Seems that I should be able to do all of this with something like `map[string]*spiffebundle.Bundle`? When considering the possibility of having a similar set of operations on *Bundle itself, the Set struct and functions feel a little confusing/overkill to me, but perhaps I'm missing something.

Reply:
The raw map feels unwieldy to pass around to me. Further, and most importantly, I needed a type that could provide `spiffex509.RootStore`, `spiffetls.SVIDStore`, `spiffetls.PeerStore` over a set of bundles organized by trust domain id.

If Set is the wrong name, I'm happy for alternatives...